Include/internal/pycore_object.h

```diff
@@ -280,6 +289,68 @@ extern PyStatus _PyObject_InitState(PyInterpreterState *interp);
 extern void _PyObject_FiniState(PyInterpreterState *interp);
 extern bool _PyRefchain_IsTraced(PyInterpreterState *interp, PyObject *obj);
 
+static inline void
```

Comment:
Can we #define `_Py_INCREF_TYPE` as `Py_INCREF` for the GIL build. I'm concerned that we will overwhelm the inlining capabilities of some of the compilers given that these macros/functions are used in functions that are themselves inlined.